### PR TITLE
doc: use full bsp name for html documents

### PR DESCRIPTION
**Pull request by jonas-rem**
Mainline BSPs and vendor BSPs can have the same PD name as hinted in https://github.com/phytec/doc-bsp-yocto/pull/201#discussion_r1672213651. It is important to differentiate in the name of the BSP documents.

The Phytec Download section differentiates between Mainline and Vendor BSPs clearly. However, if customers browse the documentation it it not clear from the document overview what kind of BSP is described.

**Do not merge yet:**
- [x] This PR changes the HTML Document name and therefore the links change. Merging this PR has to be coordinated with changing the Website links.

### CODE REVIEW

File: source/bsp/imx8/imx8mp/imx8mp.rst [hunk 1]
```diff
@@ -13,7 +13,7 @@ HEAD
    head
 
 Mainline HEAD
-=============
+=================
```
Comment (mschwan-phytec) [inline source/bsp/imx8/imx8mp/imx8mp.rst:16]: Unnecessary change I guess?

**Review by mschwan-phytec — COMMENTED** [review]
Looks good overall. One small thing I noticed...

**Review by ymoog — DISMISSED** [review]

File: source/bsp/imx8/imx8mp/imx8mp.rst (hunk 1 above)
Comment (jonas-rem) [reply source/bsp/imx8/imx8mp/imx8mp.rst]: Right, I removed it from this PR.

**Review by ymoog — APPROVED** [review]

**Review by mschwan-phytec — APPROVED** [review]

**Review by tremmet — APPROVED** [review]
I can change the links to the documentation.
